torch/distributed/_composable/fsdp/_fsdp_init.py

```diff
@@ -16,3 +19,14 @@ def _normalize_device(device: DeviceLikeType) -> torch.device:
         return torch.device(device)
     else:
         raise TypeError(f"Invalid type for device {device}: {type(device)}")
+
+
+def _init_default_fully_shard_mesh(device_type: str) -> DeviceMesh:
+    """The default fully-shard mesh shards over the global mesh."""
+    default_pg = dist.distributed_c10d._get_default_group()
```

Comment:
are we always assuming there's going to be a default pg? I think if user pass does not pass in `mesh` and does not init a world pg by themselves, FSDP should also work?

i.e. currently inside `DeviceMesh` constructor if user don't initialize default pg, device mesh will try to initialize the default pg from the torchrun envvars and set the corresponding cuda device [here](https://github.com/pytorch/pytorch/blob/main/torch/distributed/device_mesh.py#L227-L249), if user init a default pg already it would try to reuse that default pg.

Maybe for the default init case we can simply utilizes this logic instead of relying on the assumption that a default pg is there?

Reply:
I added the following for now:
```
if not dist.distributed_c10d.is_initialized():
    dist.distributed_c10d.init_process_group()
```

Reply:
ill take the contrary position for fun.  why should we be so accomodating to people that don't initialize their mesh/pg?  Do we find this is a really important convenience, or can we simplify our code and standardize peoples' workflows more by just making everyone do an init and then pass a mesh?

Reply:
This is actually a good point, I'm ok with either ways, if we want to force a init to think about how to initialize communicator, we can make it a non-optional arg?